### PR TITLE
FontAwesome fa-layers-counter positions

### PR DESCRIPTION
**Pull request by NayeBeckham**
Solves [#309](https://github.com/FortAwesome/angular-fontawesome/issues/309)

**What's the problem**
There was a missing prop to set the position of `fa-layers-counter`. 
The position options are: `fa-layers-bottom-left`, `fa-layers-bottom-right`, `fa-layers-top-left` and `fa-layers-top-right`.
It was possible to have access to these positions through `[classes]="['fa-layers-bottom-left']"`, but input should be a better solution and more comfortable to the user.

**Why we should fix it**
It should be fixed because it is a new and necessary feature to the user, who is going to be able now to set a position in its `fa-layers-counter` component in the easiest way.

**How I fixed it**
First of all, I created an `@Input() position?: 'bottom-right' | 'bottom-left' | 'top-right' | 'top-left'; ` in the FaLayersCounterComponent.
This input should be like this in fa-layers-counter component, e.g: `[position]="'bottom-left'"`
The curious thing about this input is it needs to be in the `span` tag inside `fa-layers-counter` component. Otherwise, the position input wasn't working correctly. The `span` tag isn't visible in the HTML file, it is an inner HTML child in the component.
So, the build params function looks like this:
```
if (this.classes != null) {
      classes.push(...this.classes);
}
if (this.position != null) {
      classes.push(`fa-layers-${this.position}`);
}
```
`this.classes` is an array inside the `params` variable, it is needed to have all the classes that we user wants and push the new ones for the position.
Next, for a visual demonstration, I added native HTML `select` so the user can be able to choose and select one position and see how it works.
Finally, I did my unit test, I ran `yarn lint` and `yarn test` and everything was working fine.

Here is a video to show the visual demonstration and functionality:

https://user-images.githubusercontent.com/78670199/147312875-9cead2cb-34a6-46c0-abe7-9b9b75b8ba18.mov

Thank you, feel free to tell me anything.

### CODE REVIEW

File: projects/demo/src/app/app.component.html [hunk 1]
```diff
@@ -81,18 +81,44 @@ <h3>Layers</h3>
...
 </div>
-<p>Slide and increase the number.</p>
-<input
-  type="range"
-  [min]="1000"
-  [max]="2000"
-  [value]="notificationsCounter"
-  (input)="notificationsCounter = $event.target.value"
-/>
+<mat-grid-list cols="2" rowHeight="20px">
```
Comment (devoto13) [inline projects/demo/src/app/app.component.html:84]: I'm a bit conflicted about bringing in Angular Material as a dependency. I understand that you try to make the demo app look nicer, but IMO it is better to use simple HTML in the demo app for two reasons:

- Less styles and simpler code in the demo app should make it easier to spot and troubleshoot potential issues as there are less variables at play.
- Smaller maintenance effort moving forward as there are less dependencies to take care of.

We can probably open a new issue to discuss how to improve look and feel of the demo website, but I would prefer to keep this PR focused on the position input and remove code which is not directly related to it. I think here it should suffice to have a borderless table and native HTML select element.

File: package.json [hunk 2]
```diff
@@ -70,6 +70,7 @@
     "zone.js": "~0.11.4"
   },
   "dependencies": {
+    "@angular/cdk": "^13.1.1",
```
Comment (devoto13) [inline package.json:73]: Even if we decide to keep Angular Material, it should be added to the devDependencies. It is only used in the `demo` website and should not be a production dependency of `@fortawesome/angular-fontawesome` NPM package. Otherwise every consumer of the library will end up installing it and unnecessarily bloating their Node modules.

File: src/lib/layers/layers-counter.component.ts [hunk 3]
```diff
@@ -8,23 +8,34 @@ import { FaLayersComponent } from './layers.component';
...
+    class: 'fa-layers-counter',
   },
 })
-export class FaLayersCounterComponent implements OnChanges {
+export class FaLayersCounterComponent implements OnChanges, OnInit {
   @Input() content: string;
   @Input() title?: string;
   @Input() styles?: Styles;
   @Input() classes?: string[] = [];
+  @Input() position?: string;
```
Comment (devoto13) [inline src/lib/layers/layers-counter.component.ts:19]: How about changing the type from `string` to the string union of `'bottom-right' | 'bottom-left' | 'top-right' | 'top-left'`? This way user will get a compilation error if they make a typo in the value and autocompletion as they pick it resulting in a better user experience.

File: src/lib/shared/models/props.model.ts [hunk 4]
```diff
@@ -29,4 +29,6 @@ export interface FaProps {
   symbol?: FaSymbol;
   style?: Styles;
   stackItemSize?: '1x' | '2x';
+  position?: 'bottom-right' | 'bottom-left' | 'top-right' | 'top-left';
```
Comment (devoto13) [inline src/lib/shared/models/props.model.ts:32]: This does not seem to be necessary and can probably be reverted.

File: src/lib/layers/layers-counter.component.ts [hunk 5]
```diff
@@ -8,23 +8,34 @@ import { FaLayersComponent } from './layers.component';
   selector: 'fa-layers-counter',
   template: '',
   host: {
-    class: 'ng-fa-layers-counter',
+    class: 'fa-layers-counter',
```
Comment (devoto13) [inline src/lib/layers/layers-counter.component.ts:11]: Why is this change needed?

File: src/lib/layers/layers-counter.component.ts [hunk 6]
```diff
@@ -8,23 +8,34 @@ import { FaLayersComponent } from './layers.component';
...
 
   @HostBinding('innerHTML') renderedHTML: SafeHtml;
 
-  constructor(@Optional() private parent: FaLayersComponent, private sanitizer: DomSanitizer) {
+  constructor(@Optional() private parent: FaLayersComponent, private sanitizer: DomSanitizer, private renderer: Renderer2, private elementRef: ElementRef) {
     faWarnIfParentNotExist(this.parent, 'FaLayersComponent', this.constructor.name);
   }
 
+  ngOnInit() {
+    this.renderer.removeClass(this.elementRef.nativeElement, 'fa-layers-counter');
```
Comment (devoto13) [inline src/lib/layers/layers-counter.component.ts:28]: Why was this change needed?

File: src/lib/layers/layers-counter.component.ts [hunk 7]
```diff
@@ -8,23 +8,34 @@ import { FaLayersComponent } from './layers.component';
...
+  }
+
   ngOnChanges(changes: SimpleChanges) {
     if (changes) {
+      if ('position' in changes) {
+        if (changes.position.currentValue !== null) {
+          this.classes.pop();
```
Comment (devoto13) [inline src/lib/layers/layers-counter.component.ts:35]: Wouldn't this behave incorrectly when user attempts to pass a custom class using the `classes` input. E.g. `<fa-layers-counter [classes]="['my-custom-class']" position="top-right"></fa-layers-counter>` will not have `my-custom-class`. 

It is also pretty dangerous to mutate the array passed in as an input as it may have unintended effects on repeatable render passes, I guess that's why you had to add that `.pop()` call at the first place.

IMO it would be safer and cleaner to build a new classes array with all relevant classes in the `buildParams` method. Something like below. WDYT?

```ts
const classes = [];
if (this.classes != null) {
  classes.push(...this.classes);
}
if (this.position != null) {
  classes.push(`fa-layers-${changes.position.currentValue}`);
}
return {
      title: this.title,
      classes: classes,
      styles: this.styles,
    };
```

**Review by devoto13 — COMMENTED** [review]
Thanks for the PR! I think it is a very good start. I've left some comments and questions, please let me know if something is not clear.

File: projects/demo/src/app/app.component.html (hunk 1 above)
Comment (NayeBeckham) [reply projects/demo/src/app/app.component.html:84]: I would make it native HTML for now. :) Thank you!

File: src/lib/layers/layers-counter.component.ts (hunk 3 above)
Comment (NayeBeckham) [reply src/lib/layers/layers-counter.component.ts:19]: Ok, got it!

File: src/lib/layers/layers-counter.component.ts (hunk 5 above)
Comment (NayeBeckham) [reply src/lib/layers/layers-counter.component.ts:11]: Sorry, my bad. I think it could back as normal without problem!

File: src/lib/layers/layers-counter.component.ts (hunk 6 above)
Comment (NayeBeckham) [reply src/lib/layers/layers-counter.component.ts:28]: The thing here, is we don't need the position class to be in `fa-layers-counter`. That's why I was removing it from the component because I was having some trouble. But, it can also work without the removeClass function.

File: src/lib/layers/layers-counter.component.ts (hunk 7 above)
Comment (NayeBeckham) [reply src/lib/layers/layers-counter.component.ts:35]: You're right. Thank you for the feedback!

**Review by devoto13 — APPROVED** [review]
LGTM
Thank you for implementing this feature!
